### PR TITLE
Update docker client to latest (8.5.0)

### LINKED ISSUE #333: Update docker-client
## Description

Update `docker-client` to latest version.

## How to reproduce

n.a.

## What do you expect

Set to `8.4.0`

## What happened instead

Currently set to `5.0.1` in `pom.xml`

## Software:

- docker-maven-plugin version: HEAD

### PR DESCRIPTION
**Pull request by unicolet**
Fixes #333 preparatory step for when docker client ships with https://github.com/spotify/docker-client/pull/744

@mattnworb it's me again ;-)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -125,7 +125,7 @@
     <dependency>
       <groupId>com.spotify</groupId>
       <artifactId>docker-client</artifactId>
-      <version>5.0.1</version>
+      <version>8.4.0</version>
```
Comment (davidxia) [inline pom.xml]: I'm assuming you're going to use 8.5.0 when it's released?

**Comment by mattnworb** [comment]
I'm in the process of release docker-client:8.5.0 now

**Comment by unicolet** [comment]
Yes thanks

**Comment by mattnworb** [comment]
8.5.0 is released; i think it sometimes takes maven central a few hours to be updated.

**Comment by unicolet** [comment]
Bumped `docker-client` to `8.5.0` :-)

Done a test locally and the connection pool problem is gone (and the rest still works :stuck_out_tongue: ), yay!

**Comment by mattnworb** [comment]
Thanks!

**Comment by unicolet** [comment]
@mattnworb thanks, do you think you can ship a new version of the plugin with this PR? Thanks again :-)

**Comment by mattnworb** [comment]
@unicolet I just released 0.4.14.

**Comment by unicolet** [comment]
@mattnworb THANK YOU!

**Pull request merged**